src/silo/common/fasta_reader.test.cpp

```diff
@@ -0,0 +1,375 @@
+#include <gtest/gtest.h>
+#include <string>
+#include <unordered_map>
+
+#include "silo/common/fasta_format_exception.h"
+#include "silo/common/fasta_reader.h"
+#include "silo/preprocessing/preprocessing_config.h"
+
+TEST(FastaReader, shouldReadFastaFile) {
+   const silo::InputDirectory input_directory{"./testBaseData/"};
+   const silo::OutputDirectory output_directory{"./build/"};
+   const silo::MetadataFilename metadata_filename{"small_metadata_set.tsv"};
+   const silo::SequenceFilename sequence_filename{"small_sequence_set.fasta"};
+   auto config = silo::PreprocessingConfig(
+      input_directory, output_directory, metadata_filename, sequence_filename
+   );
+   silo::FastaReader in_file(config.sequence_file);
+
+   std::string key;
+   std::string genome;
+   EXPECT_TRUE(in_file.nextKey(key));
+   EXPECT_EQ(key, "EPI_ISL_1408408");
+
+   EXPECT_TRUE(in_file.next(key, genome));
+   EXPECT_EQ(key, "EPI_ISL_1749899");
+   EXPECT_EQ(
+      genome,
+      "------------------------------"
+      "ACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCA"
```

Comment:
We could as well test this with an artificial sequence file that has a shorter genome. This is a unit test, so we only need data that has a correct structure.